crates/bevy_reflect/derive/src/result_sifter.rs

```diff
@@ -0,0 +1,40 @@
+use perfect_derive::perfect_derive;
+
+/// Helper struct used to process an iterator of `Result<Vec<T>, syn::Error>`,
+/// combining errors into one along the way.
+#[perfect_derive(Default)]
```

Comment:
Any reason we need to bring in a new dependency for this? I'm not super against adding dependencies, but I feel like in this case we could just settle for the 8-line `Default` impl.

Reply:
Whichever way you think is a cleaner solution

Reply:
IMO an additional dependency to avoid writing a manual `Default` implementation is not worth it, at least for libraries. Moreover we already have many manual implementations to work around the fact that `derive`s put unnecessary bounds on generics, so using `perfect_derive` here would be inconsistent.